samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -410,9 +442,9 @@ static int presence_detected_get(struct bt_mesh_sensor_srv *srv,
 				 struct bt_mesh_sensor_value *rsp)
 {
 	int err = bt_mesh_sensor_value_from_micro(sensor->type->channels[0].format,
-						  pres_detect * 1000000LL, rsp);
+						  BASE_UNITS_TO_MICRO(pres_detect), rsp);
```

Comment:
As I wrote before, boolean `True` is just not zero. Behavior of this implementation is not predictable.

Reply:
Sorry Alex, your previous comment got away from me. How should I fix this? typecast? I guess the (bool) * 1000000LL was done by @ludvigsj initially.

Reply:
`BASE_UNITS_TO_MICRO(!!pres_detect)` is enough here.

Reply:
Nop it should be something like:
```
BASE_UNITS_TO_MICRO(pres_detect ? 1 : 0)
```
I faced that some compilers with max optimization skipped adapting `True` to `1` and just used any digit for it if it is not zero.

Reply:
They optimize out `!!`? I've never seen that I think, I thought `!!` was commonly used to force 1 or 0? But sure, your way is 100% unambiguous.

Reply:
Nop. `!!` manages perfectly with conversion of any numeric type to boolean. This is the opposite situation. Boolean to numeric.